### ISSUE TITLE
Ability to enable/disable the compatibility mode

### ISSUE THREAD

**Issue by slemeur** [containers/podman-desktop#1447]
### Is your enhancement related to a problem? Please describe

As a user starting with Podman Desktop, I can see a warning message about the Docker Compatibility mode - but I have no way to directly enable it from the application.

### Describe the solution you'd like

Along with the work on the Compatibility mode done in #905, we would like to enable the user to turn it on/off. 

We should have that ability in the Preferences>Settings.

From the dashboard, as a user I would like the ability to fix this warning: a link to the setting would do the trick to start with. (later would be part of the onboarding sequence)
![Image](https://user-images.githubusercontent.com/1636769/218779082-f047a00e-c7fa-4ba5-a1dd-cc0984dcf136.png)

We should also handle the case where the user started podman desktop, started the podman-machine and then stopped docker. It should still be possible to enable the docker-helper.

**Comment by cdrage** [comment]
Now that I fully understand what you mean @slemeur I can outline what needs to be done as it will increase the scope of the implementation:

**On / Off button location:** It should be under the Podman "provider" as we will need to see the output of the enabling / disabling of the compatibility mode. Since we'll be running commands such as `podman-mac-helper` or aliasing.

### Different OS implementations

macOS: 
* We'll have to detect that `podman-mac-helper` is installed, and if it's installed, we'll have to run it. 
* We also need to check if the docker socket is being ran either in /var/run/docker.sock or ~/.docker/run/docker.sock
* We can also run `ln -s /var/run/podman.sock /var/run/docker.sock` ?

Linux: 
* Check socket locations and run `ln -s /var/run/podman.sock /var/run/docker.sock`

Windows:
* Unsure at the moment what command would correctly alias it.. current instructions is to just restart and it'll alias the socket with `podman machine restart`. But there's no option on `podman machine` to enable / disable the aliasing. I need to research this part.


### Misc

* Only be able to enable docker compatibility if docker desktop is stopped / not running

Is this all correct @benoitf @slemeur / am I missing anything here?

**Comment by slemeur** [comment]
Sounds good for me!

**Comment by cdrage** [comment]
OS:

- [X] macOS https://github.com/containers/podman-desktop/pull/1697
- [ ] Windows https://github.com/containers/podman-desktop/pull/1769
- [ ] Linux

**Comment by cdrage** [comment]
Two blockers for this which are causing issues:

- Windows Docker provider is sometimes appearing and sometimes not: https://github.com/containers/podman-desktop/issues/1819
- Compatibility mode isn't working very well in Linux in dev mode and if it's compile non-flatpak: https://github.com/containers/podman-desktop/issues/1912

**Comment by flibustier7seas** [comment]
Any plans to add such functionality for Windows?

**Comment by benoitf** [comment]
on Windows, podman machine command is always trying to take the docker npipe if it's free.

on macOS the 3rd party tool requires admin privileges this is why there is an extra "installation" step

**Comment by flibustier7seas** [comment]
In some cases, I need to run Docker, but as Podman holds the pipe, Docker can't start (https://github.com/containers/podman/issues/19233). 

Currently, I have to stop Podman, run Docker, and only after that run Podman, and it's quite annoying.

So, it would be nice to have the option to disable `Compatibility mode` on Windows.
